src/arm/decoder.c

```diff
@@ -488,3 +488,43 @@ int ARMDisassemble(struct ARMInstructionInfo* info, uint32_t pc, char* buffer, i
 	buffer[blen - 1] = '\0';
 	return total;
 }
+
+uint32_t ARMResolveMemoryAccess(struct ARMInstructionInfo* info, struct ARMRegisterFile* regs, uint32_t pc) {
+	uint32_t address = 0;
+	int32_t offset = 0;
+	if (info->memory.format & ARM_MEMORY_REGISTER_BASE) {
+		if (info->memory.baseReg == ARM_PC && info->memory.format & ARM_MEMORY_IMMEDIATE_OFFSET) {
+			address = pc & 0xFFFFFFFC;
```

Comment:
This is incorrect for Thumb. Also its force-aligned already so you don't need to mask it.

Reply:
Oh right, I don't need to account for the switch-to-Thumb low bit on the _current_ pc value.